extensions/resteasy-reactive/quarkus-resteasy-reactive-jackson/deployment/src/main/java/io/quarkus/resteasy/reactive/jackson/deployment/processor/ResteasyReactiveJacksonProcessor.java

```diff
@@ -122,10 +124,13 @@ AdditionalBeanBuildItem beans() {
     }
 
     @BuildStep
-    void additionalProviders(List<JacksonFeatureBuildItem> jacksonFeatureBuildItems,
+    void additionalProviders(ContextResolversBuildItem contextResolversBuildItem,
+            List<JacksonFeatureBuildItem> jacksonFeatureBuildItems,
             BuildProducer<MessageBodyReaderBuildItem> additionalReaders,
             BuildProducer<MessageBodyWriterBuildItem> additionalWriters) {
-        boolean applicationNeedsSpecialJacksonFeatures = jacksonFeatureBuildItems.isEmpty();
+        boolean specialJacksonFeaturesUsed = !jacksonFeatureBuildItems.isEmpty();
+        boolean hasObjectMapperContextResolver = contextResolversBuildItem.getContextResolvers().getResolvers()
+                .containsKey(ObjectMapper.class);
```

Comment:
The key is a `Class` or an `ObjectMapper` instance?

Reply:
The former